components/ui/checkbox.tsx

```diff
@@ -12,13 +16,40 @@ interface CheckboxProps {
   iconSize?: number;
 }
 
-const AnimatedCheck = Animated.createAnimatedComponent(Check);
+interface AnimatedCheckProps {
+  value: boolean;
+  size: number;
+  className: string;
+  opacity: number;
+}
+
+const AnimatedCheck = Animated.createAnimatedComponent(
+  React.forwardRef(
+    (
+      { value, size, className, opacity, ...props }: AnimatedCheckProps,
+      ref: React.ForwardedRef<any>
+    ) => {
+      return (
+        <View ref={ref} style={{ opacity }}>
+          <Check size={size} className={className} {...props} />
+        </View>
+      );
+    }
+  )
+);
 
 const Checkbox = React.forwardRef<
   React.ElementRef<typeof Pressable>,
   Omit<React.ComponentPropsWithoutRef<typeof Pressable>, 'onPress'> &
     CheckboxProps
 >(({ className, value, onChange, iconClass, iconSize = 16, ...props }, ref) => {
+  const opacity = useSharedValue(0);
+
+  opacity.value = withTiming(value === true ? 1.0 : 0.0, {
+    duration: 250,
+    easing: Easing.inOut(Easing.ease),
+  });
```

Comment:
I don't think updating the shared value on each re-render is a good practice. I think the opacity should be updated in a useAnimatedStyle from a value from useDerivedValue

Reply:
I can see how `useDerived` value would be better here, but I'll resubmit with your solution if you prefer that one

Reply:
You're right I need to be updating the opacity.value but not the value itself. I've undone this in the latest version